ios/Classes/FlutterSoundPlugin.m

```diff
@@ -282,7 +282,7 @@ - (void)startRecorder
   if ([path class] == [NSNull class]) {
     audioFileURL = [NSURL fileURLWithPath:[GetDirectoryOfType_FlutterSound(NSCachesDirectory) stringByAppendingString:defaultExtensions[codec] ]];
   } else {
-    audioFileURL = [NSURL fileURLWithPath: [GetDirectoryOfType_FlutterSound(NSCachesDirectory) stringByAppendingString:path]];
```

Comment:
So you don't like the cache directory? How about supporting options to choose cache or file?

Reply:
My idea is that the recorded file will be stored in the cache directory if the user does not specify a file path.
And it will be stored in the provided path if the user specify a path.

I really think that it is not necessary for the user to specify just a temporary file name and not the full path. Of course we can still do that for backward compatibility. Just tell me.

Reply:
Oh! I think your idea is absolutely great~